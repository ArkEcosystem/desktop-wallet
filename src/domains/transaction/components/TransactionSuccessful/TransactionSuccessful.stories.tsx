import React from "react";

import { TransactionSuccessful } from "./TransactionSuccessful";

export default { title: "Domains / Transaction / Components / TransactionSuccessful" };

export const Default = () => (
	<div className="max-w-xl py-16 mx-auto">
		<TransactionSuccessful>
			<div />
		</TransactionSuccessful>
	</div>
);
