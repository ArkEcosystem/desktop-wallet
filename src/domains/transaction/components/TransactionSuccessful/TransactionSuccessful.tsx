import { images } from "app/assets/images";
import { Address } from "app/components/Address";
import { Circle } from "app/components/Circle";
import { Clipboard } from "app/components/Clipboard";
import { Icon } from "app/components/Icon";
import { Label } from "app/components/Label";
import { TransactionDetail } from "app/components/TransactionDetail";
import React from "react";

const { TransactionSuccessfulBanner } = images.transaction.common;

export const TransactionSuccessful = ({ children }: { children: React.ReactNode }) => (
	<section data-testid="TransactionSuccessful" className="space-y-8">
		<div>
			<h1 className="mb-0">Transaction Successful</h1>
			<div className="grid grid-flow-row gap-2">
				<div className="w-full my-10">
					<TransactionSuccessfulBanner className="w-full" />
				</div>
				<p className="mb-4 text-theme-neutral-dark">
					Your transaction was successfully sent. Please monitor the blockchain to ensure your transaction is
					confirmed and processed. The following is relevant information for your transaction:
				</p>
				<TransactionDetail label="ID" border={false}>
					<div className="flex items-center">
						<Clipboard>
							<Address
								addressClass="text-theme-primary"
								address="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWKAUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK"
								maxChars={32}
							/>
						</Clipboard>
						<div className="mb-1 ml-5 text-theme-primary-300">
							<Icon name="Copy" />
						</div>
					</div>
				</TransactionDetail>

				<TransactionDetail label="Block ID">
					<div className="flex items-center">
						<Clipboard>
							<Address
								addressClass="text-theme-primary"
								address="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWKAUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK"
								maxChars={32}
							/>
						</Clipboard>
						<div className="mb-1 ml-5 text-theme-primary-300">
							<Icon name="Copy" />
						</div>
					</div>
				</TransactionDetail>

				<TransactionDetail
					label="Network"
					extra={
						<div className="ml-1 text-theme-danger-500">
							<Circle className="bg-theme-background border-theme-danger-200" size="lg">
								<Icon name="Ark" width={20} height={20} />
							</Circle>
						</div>
					}
				>
					<Address address="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK" walletName={"ROBank"} />
				</TransactionDetail>

				<TransactionDetail
					extra={
						<div className="mt-2">
							<Circle avatarId="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK" />
						</div>
					}
				>
					<div className="mb-2 text-sm font-semibold text-theme-neutral-500">
						<span className="mr-1">Sender</span>
						<Label color="warning">
							<span className="text-sm">Your address</span>
						</Label>
					</div>
					<Address address="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK" walletName={"ROBank"} />
				</TransactionDetail>

				{children}
			</div>
		</div>
	</section>
);
