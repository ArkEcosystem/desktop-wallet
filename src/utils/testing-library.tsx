import { ARK } from "@arkecosystem/platform-sdk-ark";
import { Environment } from "@arkecosystem/platform-sdk-profiles";
import { render } from "@testing-library/react";
import { EnvironmentProvider } from "app/contexts";
import { i18n } from "app/i18n";
import { httpClient } from "app/services";
import { createMemoryHistory } from "history";
import React from "react";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";
import { StubStorage } from "tests/mocks";

import envFixture from "../tests/fixtures/env/data.json";

export const env = new Environment({ coins: { ARK }, httpClient, storage: new StubStorage(envFixture) });

const WithProviders: React.FC = ({ children }: { children?: React.ReactNode }) => (
	<I18nextProvider i18n={i18n}>
		<EnvironmentProvider env={env}>{children}</EnvironmentProvider>
	</I18nextProvider>
);

const customRender = (component: React.ReactElement, options: any = {}) =>
	render(component, { wrapper: WithProviders, ...options });

const renderWithRouter = (
	component: React.ReactElement,
	{ routes = ["/"], history = createMemoryHistory({ initialEntries: routes }), withProviders = true } = {},
) => {
	const RouterWrapper = ({ children }: { children: React.ReactNode }) =>
		withProviders ? (
			<WithProviders>
				<Router history={history}>{children}</Router>
			</WithProviders>
		) : (
			<Router history={history}>{children}</Router>
		);

	return {
		...customRender(component, { wrapper: RouterWrapper }),
		history,
	};
};

export * from "@testing-library/react";

export { customRender as render, renderWithRouter };
