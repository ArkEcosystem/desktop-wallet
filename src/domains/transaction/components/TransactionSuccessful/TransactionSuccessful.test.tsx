import React from "react";
import { render } from "testing-library";

import { TransactionSuccessful } from "./TransactionSuccessful";

describe("TransactionSuccessful", () => {
	it("should render", () => {
		const { asFragment } = render(
			<TransactionSuccessful>
				<div />
			</TransactionSuccessful>,
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
