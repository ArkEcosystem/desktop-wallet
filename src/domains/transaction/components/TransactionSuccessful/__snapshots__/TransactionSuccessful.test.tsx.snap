// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionSuccessful should render 1`] = `
<DocumentFragment>
  <section
    class="space-y-8"
    data-testid="TransactionSuccessful"
  >
    <div>
      <h1
        class="mb-0"
      >
        Transaction Successful
      </h1>
      <div
        class="grid grid-flow-row gap-2"
      >
        <div
          class="w-full my-10"
        >
          <svg
            class="w-full"
          >
            transaction_successful.svg
          </svg>
        </div>
        <p
          class="mb-4 text-theme-neutral-dark"
        >
          Your transaction was successfully sent. Please monitor the blockchain to ensure your transaction is confirmed and processed. The following is relevant information for your transaction:
        </p>
        <div
          class="flex items-center py-6 true "
        >
          <div
            class="flex-1 space-y-2"
          >
            <div
              class="text-sm font-semibold text-theme-neutral"
            >
              ID
            </div>
            <div
              class="font-semibold"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="inline-block cursor-pointer"
                  data-testid="clipboard__wrapper"
                >
                  <div
                    class="inline-block truncate align-middle"
                  >
                    <span
                      class="sc-AxjAm StDqN text-theme-primary font-semibold text-base"
                      data-testid="address__wallet-address"
                    >
                      AUexKjGtgsSpVz…MM6vJ6znEVTQWK
                    </span>
                  </div>
                </div>
                <div
                  class="mb-1 ml-5 text-theme-primary-300"
                >
                  <div
                    class="sc-AxiKw dzBdqj"
                    height="1em"
                    width="1em"
                  >
                    <svg>
                      copy.svg
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex items-center py-6 border-t border-dashed border-theme-neutral-300 "
        >
          <div
            class="flex-1 space-y-2"
          >
            <div
              class="text-sm font-semibold text-theme-neutral"
            >
              Block ID
            </div>
            <div
              class="font-semibold"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="inline-block cursor-pointer"
                  data-testid="clipboard__wrapper"
                >
                  <div
                    class="inline-block truncate align-middle"
                  >
                    <span
                      class="sc-AxjAm StDqN text-theme-primary font-semibold text-base"
                      data-testid="address__wallet-address"
                    >
                      AUexKjGtgsSpVz…MM6vJ6znEVTQWK
                    </span>
                  </div>
                </div>
                <div
                  class="mb-1 ml-5 text-theme-primary-300"
                >
                  <div
                    class="sc-AxiKw dzBdqj"
                    height="1em"
                    width="1em"
                  >
                    <svg>
                      copy.svg
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex items-center py-6 border-t border-dashed border-theme-neutral-300 "
        >
          <div
            class="flex-1 space-y-2"
          >
            <div
              class="text-sm font-semibold text-theme-neutral"
            >
              Network
            </div>
            <div
              class="font-semibold"
            >
              <div
                class="inline-block truncate align-middle"
              >
                <span
                  class="text-theme-text max-w-24 flex-auto truncate mt-4 mr-1 font-semibold text-base"
                  data-testid="address__wallet-name"
                >
                  ROBank
                </span>
                <span
                  class="sc-AxjAm StDqN text-theme-neutral-light font-semibold text-base"
                  data-testid="address__wallet-address"
                >
                  AUexKj…EVTQWK
                </span>
              </div>
            </div>
          </div>
          <div
            class="ml-1 text-theme-danger"
          >
            <div
              class="sc-AxirZ cZokSJ bg-theme-background border-theme-danger-light"
            >
              <div
                class="sc-AxiKw dklFyP"
                height="20"
                width="20"
              >
                <svg>
                  ark.svg
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex items-center py-6 border-t border-dashed border-theme-neutral-300 "
        >
          <div
            class="flex-1 space-y-2"
          >
            <div
              class="font-semibold"
            >
              <div
                class="mb-2 text-sm font-semibold text-theme-neutral"
              >
                <span
                  class="mr-1"
                >
                  Sender
                </span>
                <div
                  class="sc-AxhCb jwGgjn"
                  color="warning"
                >
                  <span
                    class="text-sm"
                  >
                    Your address
                  </span>
                </div>
              </div>
              <div
                class="inline-block truncate align-middle"
              >
                <span
                  class="text-theme-text max-w-24 flex-auto truncate mt-4 mr-1 font-semibold text-base"
                  data-testid="address__wallet-name"
                >
                  ROBank
                </span>
                <span
                  class="sc-AxjAm StDqN text-theme-neutral-light font-semibold text-base"
                  data-testid="address__wallet-address"
                >
                  AUexKj…EVTQWK
                </span>
              </div>
            </div>
          </div>
          <div
            class="Avatar__AvatarWrapper-sc-1vsy2s2-0 ikUhMz"
            data-testid="Avatar"
          >
            <div
              class="w-full h-full"
            >
              <img
                alt="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK"
                src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(233, 30, 99)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"55\\" cx=\\"0\\" cy=\\"30\\" fill=\\"rgb(255, 87, 34)\\"/><circle r=\\"45\\" cx=\\"80\\" cy=\\"60\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"10\\" cy=\\"50\\" fill=\\"rgb(63, 81, 181)\\"/></svg>"
                title="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK"
              />
            </div>
          </div>
        </div>
        <div />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
